### PR TITLE
fix(Core/Loot): Properly handle referenced group loot templates.

### LINKED ISSUE #5781: Prospecting Thorium doesn't always yield a gem
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [X] enUS
- [ ] other (specify)

##### FACTION

- [X] Alliance
- [X] Horde

##### CONTENT PHASE:

- [X] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Original issue from Chromiecraft
[https://github.com/chromiecraft/chromiecraft/issues/600](https://github.com/chromiecraft/chromiecraft/issues/600)
Occasionally when prospecting thorium you get a blank window instead of actually getting a gem. 

##### EXPECTED BLIZZLIKE BEHAVIOUR:

[https://wowwiki-archive.fandom.com/wiki/Prospecting](https://wowwiki-archive.fandom.com/wiki/Prospecting)
As of patch 2.1.0 prospecting always yields at least 1 gem

##### STEPS TO REPRODUCE THE PROBLEM:

1. .additem 10620 300
2. set your prospecting to a number key and start tapping
3. 

##### EXTRA NOTES:
Prospected 300 of all ores that can be prospected, and I've only found thorium to give a blank.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/6947789622de45479d2311fd5d1635af3a957b5d

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98393265-prospecting-thorium-doesn-t-always-yield-a-gem?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #5781

## Issues Addressed:

- Closes #5781
- closes https://github.com/chromiecraft/chromiecraft/issues/600
- Might close other tickets about empty loot tables using references.

## Tests Performed:

- Tested ingame with prospecting.

## How to Test the Changes:

1. `.additem 10620 300`
2. `.learn 31252`
3. set your prospecting to a number key and start tapping

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
@UltraNix still failing, please check the build

**Comment by UltraNix** [comment]
Weird - that SQL file should be fine. `Reference` is now no longer unsingned, which is changed above that chnage to thorium loot.

**Comment by FrancescoBorzi** [comment]
still failing :(

**Comment by UltraNix** [comment]
So I don't know how to fix it. SQL is okay.

**Comment by Kitzunu** [comment]
in DB `reference` of prospecting_loot_template is unsigned. that's why you cant go negative

**Comment by UltraNix** [comment]
But as you can see I changed that.

**Comment by FrancescoBorzi** [comment]
what's the status of this?

**Comment by UltraNix** [comment]
Now it's ready to be merged.

File: src/server/game/Loot/LootMgr.cpp [hunk 1]
```diff
@@ -144,7 +147,7 @@ uint32 LootStore::LoadLootTable()
 
         uint32 entry               = fields[0].GetUInt32();
         uint32 item                = fields[1].GetUInt32();
-        uint32 reference           = fields[2].GetUInt32();
+        int32  reference           = fields[2].GetInt32();
```
Comment (FrancescoBorzi) [inline src/server/game/Loot/LootMgr.cpp:150]: @UltraNix have you checked that there are no entries that are too big for `int32` which previously fit in `uint32` ?

Comment (UltraNix) [reply src/server/game/Loot/LootMgr.cpp:150]: But it's still 4 bytes. Only one entry now is signed.

Comment (FrancescoBorzi) [reply src/server/game/Loot/LootMgr.cpp:150]: @UltraNix yes... which means `uint32` can hold bigger numbers than `int32` :D

Comment (UltraNix) [reply src/server/game/Loot/LootMgr.cpp:150]: Nothing wrong happens here.

**Pull request merged**